java/ql/src/experimental/Security/CWE/CWE-918/SSRF.qhelp

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+
+  <overview>
+    <p>Server Side Request Forgery (SSRF) is an attack vector that abuses an application to interact with the internal/external network or the machine itself, frequently through the attack vector of URL mishandling.</p>
+    <p>In an SSRF attack, the attacker can abuse functionality on the server to read or update internal resources, e.g. cloud server meta-data, internal REST interface, and files.</p>
+    <p>By providing unexpected URLs, attackers can trick the application to send malicious requests, possibly bypassing access controls such as firewalls that prevent the attackers from accessing those URLs directly. The server can be used as a proxy to conduct port scanning of hosts in internal networks, use other URLs such as that can access documents on the system (using file:// or jar://), or use other content protocols such as gopher:// or tftp://, or invoke internal REST APIs.</p>
+    <p>This query detects the following two scenarios:</p>
+    <ol>
+      <li>Java networking uri.openConnection() and its derived uri.openStream(), which is a shorthand for openConnection().getInputStream(), from remote source.</li>
```

Comment:
```suggestion
      <li>Java networking URL.openConnection() and its derived URL.openStream(), which is a shorthand for openConnection().getInputStream(), from remote source.</li>
```
To make it more clear to which class the methods belong

Might also be good to rename the local variables in the code examples from `uri` -> `url`

Reply:
Thanks @Marcono1234 . This is a very good point. I've added a sanitizer and relevant test cases to reduce false positives. And I've made text changes as well.